code/controllers/subsystem/mapping.dm

```diff
@@ -181,7 +181,7 @@ var/datum/subsystem/mapping/SSmapping
 	var/datum/map_config/VM = global.config.maplist[pickedmap]
 	message_admins("Randomly rotating map to [VM.map_name]")
 	. = changemap(VM)
-	if (.)
+	if (. && VM.map_name != config.map_name)
```

Comment:
Have you confirmed that these line up, map_name is (or was) the name of the .dm file, not the dmm file.